pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java

```diff
@@ -205,10 +205,10 @@ private static void createLeadControllerResourceIfNeeded(String helixClusterName
     if (resourceConfig == null) {
       resourceConfig = new ResourceConfig(LEAD_CONTROLLER_RESOURCE_NAME);
     }
-    // Set RESOURCE_ENABLED to false if it's absent in resource config
-    if (resourceConfig.getSimpleConfig(LEAD_CONTROLLER_RESOURCE_ENABLED_KEY) == null) {
-      resourceConfig.putSimpleConfig(LEAD_CONTROLLER_RESOURCE_ENABLED_KEY, Boolean.FALSE.toString());
-    }
+    // Explicitly set RESOURCE_ENABLED to true, so that the lead controller resource is always enabled.
+    // This is to help keep the behavior consistent in all clusters for better maintenance.
+    // TODO: remove the logic of handling this config in both controller and server in the next official release.
```

Comment:
Clarify "in the next official release". Do you want to remove it _after_ the next release is done, or _before_ the next release? If _before_, then you can do it now.

Also, please tag this PR for release notes and add appropriate comments that can be noted by whoever is building the release. Questions to be addressed in checkin comments:
- What is the difference in new cluster creation before and after this commit?
- How will existing clusters behave after the release when this code is installed?
- How to find out which controller is "responsible" for a table? (add link to the doc to the rest api. Doc it if it is not already there)

Add a pointer to the design doc.

Reply:
I think you have most of these already, just clarify the comments with the exact release number (e.g. this code is to be removed whenever we release 0.8 or later version of Pinot)

Reply:
Updated the description of this PR.

Reply:
Can you please clarify "in the next release"? Do you want it removed after the next release?

Reply:
It is best if you identify all the code that needs to be removed, and put a comment head saying "Remove this block of code once 0.8.0 release is done" or something like that.